### PR TITLE
feat(chloride): Add type defs for chloride

### PR DESCRIPTION
**Pull request by zman2013**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/chloride/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    function crypto_auth_verify(signed: Buffer, data: Buffer, key: Buffer): boolean;
+
+    // *** scalar multiplication ***
+    function crypto_scalarmult(secretKey: Buffer, publicKey: Buffer): Buffer;
+
+    // *** conversions ***
+    function crypto_sign_ed25519_sk_to_curve25519(secretKey: Buffer): Buffer;
+    function crypto_sign_ed25519_pk_to_curve25519(publicKey: Buffer): Buffer;
+}
+export default chloride;
```
Comment (weswigham) [inline types/chloride/index.d.ts:56]: Based on the cjs examples in the repo, this looks like it should be `export =` and not `export default`; and, if I'm not mistaken, furthermore, these are all the "low level" primitives usually available under the "api" member of that? Shouldn't the high level API also be included? And usually preferred?

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/chloride/index.d.ts (hunk 1 above)
Comment (zman2013) [reply types/chloride/index.d.ts]: 1. _this looks like it should be export = and not export default_
  At last, I removed the namespace, and exported the functions instead.
2. _these are all the "low level" primitives usually available under the "api" member of that? Shouldn't the high level API also be included?_
  Actually, these are high level apis, the low level apis are from `sodium-native`. Chloride is a compatibility layer not defining the wrapping api. These apis is from difference underlaying implementations, e.g. `sodium-chloride` or `sodium-browserify-tweetnacl`. 
   Maybe you is misled by `https://www.npmjs.com/package/sodium`. After digging the code, I found the package doesn't depend on the `node-sodium`.

Comment (weswigham) [reply types/chloride/index.d.ts]: Yeah, I was mislead - the examples in the `chloride` repo are of using `sodium` directly: https://github.com/ssb-js/chloride/blob/main/examples/low-level-api.js

**Review by weswigham — APPROVED** [review]

**Comment by zman2013** [comment]
Ready to merge

**Pull request merged**